tv-shows-app/src/components/features/shows/ShowContainer/ShowContainer.tsx

```diff
@@ -3,31 +3,38 @@ import ShowDetails from "../ShowDetails/ShowDetails";
 import ShowReviewSection from "../ShowReviewSection/ShowReviewSection";
 import { IReview, IReviewList } from "@/typings/review";
 import { useState, useEffect } from "react";
-import useSWR from "swr";
-import { getShow } from "@/app/fetchers/shows";
 import { useParams } from "next/navigation";
 import { Params } from "next/dist/shared/lib/router/utils/route-matcher";
 import { CiWarning } from "react-icons/ci";
 import {
   loadFromLocalStorage,
   saveToLocalStorage,
 } from "./ShowContainer.utils";
+import useSWRMutation from "swr/mutation";
+import { swrKeys } from "@/app/fetchers/swrKeys";
+import { loggedMutator } from "@/app/fetchers/mutators";
 
 export default function ShowContainer() {
   const { id } = useParams() as Params;
-  const {
-    data: showResponse,
-    error,
-    isValidating,
-  } = useSWR(`/api/shows/${id}`, () => getShow(id));
 
   const [reviewList, setReviewList] = useState<IReviewList>({ reviews: [] });
 
+  const { trigger, data, isMutating, error } = useSWRMutation(
```

Comment:
I think what you need here is useSWR instead of useSWRMutation. You are only fetching shows here, are you not?